plotly/offline/offline.py

```diff
@@ -49,17 +50,13 @@ def get_plotlyjs():
     plotlyjs = resource_string('plotly', path).decode('utf-8')
     return plotlyjs
 
-def get_image_download_script(caller):
+def get_image_download_script():
     '''
-    This function will return a script that will download an image of a Plotly
-    plot.
-
-    Keyword Arguments:
-    caller ('plot', 'iplot') -- specifies which function made the call for the
-        download script. If `iplot`, then an extra condition is added into the
-        download script to ensure that download prompts aren't iniitated on
-        page reloads.
+    This function will return a script that will download an image of the
+    most recently rendered Plotly plot
     '''
+    # get the name of the function that called `get_image_download_script`
+    caller = sys._getframe(1).f_code.co_name
```

Comment:
😸 I totally see how this is nifty/sneaky, but the other approach was really straightforward/simple. I'm definitely not immune to trying to do things in a *clever* way, but these things can come back to bite you. For example, it's unclear to me whether this would work correctly under the following conditions:

1. we want to mock this function in a test
2. we refactor later to use a `@decorator`

Also, if you look at the [docs for `_getframe`](https://docs.python.org/2/library/sys.html#sys._getframe), they say:

>This function should be used for internal and specialized purposes only. It is not guaranteed to exist in all implementations of Python.

Also, developers will typically use the `_*` naming convention to suggest that the function is not part of the public api!